pkg/controller/interconnect/interconnect_controller.go

```diff
@@ -676,19 +674,13 @@ func (r *ReconcileInterconnect) Reconcile(request reconcile.Request) (reconcile.
 	}
 
 	// List the pods for this deployment
-	podList := &corev1.PodList{}
-	labelSelector := selectors.ResourcesByInterconnectName(instance.Name)
-	listOps := &client.ListOptions{Namespace: instance.Namespace, LabelSelector: labelSelector}
-	err = r.client.List(context.TODO(), listOps, podList)
-	if err != nil {
-		reqLogger.Error(err, "Failed to list pods")
-		return reconcile.Result{}, err
-	}
-	podNames := getPodNames(podList.Items)
+	reqLogger.Info("Collecting Pod Status")
+	dStatus := getPodStatus(r, instance)
```

Comment:
I'm not quite following the logic here. It looks like the pod statuses are retrieved for whichever placement option is used, but then it is ignored unless placement is 'Every'? Or am I missing something?

Reply:
That was not good, agree. 
I change to use DeploymentStatus.Starting slice.  If Slice is not empty, we will continue the loop.

Reply:
@myeung18  Why? What does `starting` have to do with it?

Reply:
Starting mean there are pods that is still creating/starting and they are not in ready status. So we should continue the loop, unless there are better way to do so.

Reply:
That does not seem correct to me. Starting means a specific state. If the pod starts faster, you may never see it. After it has started, it goes away, which probably indicates a bug in your code right now. Please test thoroughly to make sure you get the right behavior.

Reply:
if the pods status is somehow changed, that means it is different from the existing podstaus in instance object, this should fall to the or condition check where both statuses are compared to determine if the status should be updated.

Reply:
Oh right, it's an `or`. So why do you need the first condition?

Reply:
The daemonset pod creation somehow is slower.  After the initial update and requeue. The controller will get exact same status data with all pods in starting state.  This event won’t get the chance to enter the if block.   The If block is not executed and the requeue is not called.  Then no status update event comes in even after all the pods are created.  Adding this starting slice check meanly wants to have that requeue get called.

Reply:
The daemonset pod creation somehow is slower.  After the initial update and requeue. The controller will get exact same status data with all pods in starting state.  This event won’t get the chance to enter the if block.   The If block is not executed and the requeue is not called.  Then no status update event comes in even after all the pods are created.  Adding this starting slice check meanly wants to have that requeue get called.

Reply:
That's not the expected flow. Instead, you should watch created daemon set objects. That would then ensure that any changes to the deployment is reflected in the OLM status.

Reply:
Changed to watch DaemonSet and removed that Starting slice check.  The Controller will now update deployment pod status if there is newer status comes: new and old status objects are compared.